recipes-devtools/libfmi/libfmi/0001-remove-expat-static-link-dependency-from-libfmi.patch

```diff
@@ -0,0 +1,133 @@
+From b85dee1d652db0b8a6b2430667ce702186253c41 Mon Sep 17 00:00:00 2001
+From: pratheekshasn <pratheeksha.s.n@ni.com>
+Date: Thu, 12 Dec 2024 16:06:06 +0530
+Subject: [PATCH] remove expat static link dependency from libfmi
+
+Signed-off-by: pratheekshasn <pratheeksha.s.n@ni.com>
```

Comment:
Your .patch file should have a complete commit summary, as though it were a patch to upstream. And it should have an Upstream-Status line to help inform the maintainers on how to handle the patch during rebases.

https://github.com/ni/meta-nilrt/blob/nilrt/master/next/docs/styleguide.md#patch-files

Reply:
I agree that having a more complete commit message describing that we prefer to use the newer installed NILRT expat, as we have done in previous releases, is a good idea. Could also mention that the expat source code contained in the fmi-library source is older, with some unpatched CVEs, and that our CVE reporting was not finding them because the build of expat was hidden inside the fmi-library build.

Regarding the upstream status, I suspect we will never want to upstream this patch. It seems that fmi-library upstream already has [a newer commit (which we are not yet including) that adds support for an external expat library](https://github.com/modelon-community/fmi-library/commit/e871e37afc92f8864ab082cfbe0cfa56c4000540#diff-c215b9faa57d7fffad7b40f69fad059471d660e881f369ba237be17bb2ca3f66). We should get that commit in the future, at which point this patch will be unnecessary. Arguably, we could base this patch on that commit now instead of what we have already, although I'll leave that up to Alex to comment. Since we expect to drop this patch next time we update libfmi, we might not care too much.

Reply:
I have made the required changes now.